src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.cs

```diff
@@ -328,31 +328,37 @@ internal OpenTextBufferManager GetBufferManager(ITextBuffer buffer)
         internal bool TryGetBufferManager(ITextBuffer buffer, out OpenTextBufferManager bufferManager)
             => _openTextBuffers.TryGetValue(buffer, out bufferManager);
 
-        public void RefreshRenameSessionWithOptionsChanged(Option<bool> renameOption, bool newValue)
+        public void RefreshRenameSessionWithOptionsChanged(SymbolRenameOptions newOptions)
         {
             AssertIsForeground();
             VerifyNotDismissed();
 
-            // Recompute the result only if the previous result was computed with a different flag
-            if (_optionSet.GetOption(renameOption) != newValue)
+            Debug.Assert(_options != newOptions);
```

Comment:
tbh, this feels unfortunate.  i'd prefer we still just do the diff check in here and execute logic if it changed.  can we not do that?  maybe rename to RefreshRenameSessionIfOptionsChanged  ?

Reply:
I don't know how to do that reasonably. Any suggestions?

Reply:
So this works:

```C#
public delegate void RecordPropertySetter<TRecord, TValue>(ref TRecord options, TValue newValue);

public static void RefreshRenameSessionWithOptionsChanged<TValue>(Options options, RecordPropertySetter<Options, TValue> setValue, TValue currentValue, TValue newValue)
    where TValue : IEquatable<TValue>
    {
        if (currentValue.Equals(newValue))
        {
            return;
        }

        setValue(ref options, newValue);
        _options = options;
    }


RefreshRenameSessionWithOptionsChanged(options, (ref Options o, bool value) => o.A = value, options.A, newValue: false);
```

but it's rather complex...